storage/diff_types.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+#include <unordered_map>
+
+namespace diff_scheme
+{
+enum class Status
+{
+  Undefined,
+  NotAvailable,
+  Available
+};
+
+struct FileInfo final
+{
+  FileInfo(uint64_t size, uint64_t version) : m_size(size), m_version(version) {}
+  uint64_t m_size;
```

Comment:
so I guess I can't do this because marking these fields as const fields implicitly removes copy-assignment operator in FileInfo, and I have to store an object of type ```NameFileInfoMap``` as a class field into storage.